lib/spack/spack/version.py

```diff
@@ -587,14 +604,14 @@ def _cmp(self, other_lookups=None):
 
         return self.version
 
+    @coerced
+    def overlaps(self, other):
```

Comment:
overlaps or intersects?

Reply:
Done in https://github.com/spack/spack/pull/35681/commits/76d4ef974669290b9d8026f8bb4fbf29c292dff2